tests/cypress/e2e/unit_tests/provisioning_gitrepo_tests_fleet.spec.ts

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright © 2023 - 2024 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import '~/support/commands';
+import * as cypressLib from '@rancher-ecp-qa/cypress-library';
+import { qase } from 'cypress-qase-reporter/dist/mocha';
+
+beforeEach(() => {
+  cy.login();
+  cy.visit('/');
+  cy.deleteAllFleetRepos();
+});
+
+
+Cypress.config();
+describe('Test GitRepo name (more than or less than) 47 characters and see bundle name trims it.', () => {
+  const branch = "master"
+  const path = "simple-chart"
+  const repoUrl = "https://github.com/rancher/fleet-test-data/"
+  const allRepoNames = new Map([
+    [103, "test-test-test-test-test-test-test-test-test-t"],
+    [104, "test-test-test-test-test-test-test-test-test-test-test-test"],
+    [106, "test-test-test-test-123-456-789-0--test-test-test-test"],
+    [105, "Test.1-repo-local-cluster"],
+    [61, "ryhhskh-123456789+-+abdhg%^/"],
+  ]);
+  allRepoNames.forEach(
+    (repoName, qase_id) => {
+      if ((qase_id === 105 || qase_id === 61)) {
+        qase(qase_id,
+          it(`Fleet-${qase_id}: Test GitRepo min or max supported characters and and not supported names (see QASE test case)`, () => {
```

Comment:
This implementation makes tests to have different qase_id but same name.
I tested this other one and I could get different names for tests:
```ts
  const repoTestData: testData[] = [
    { qase_id: 103,
      repoName: "test-test-test-test-test-test-test-test-test-t",
      test_explanation: "47 characters is NOT TRIMMED" },
    { qase_id: 104,
      repoName: "test-test-test-test-test-test-test-test-test-test-test",
      test_explanation: "59 characters is TRIMMED" },
    { qase_id: 106,
      repoName: "test-test-test-test-123-456-789-0--test-test-test-test",
      test_explanation: "54 chars is TRIMMED" },
    { qase_id: 105,
      repoName: "Test.1-repo-local-cluster",
      test_explanation: "INVALID and NORMAL characters" },
    { qase_id: 61,
      repoName: "ryhhskh-123456789+-+abdhg%^/",
      test_explanation: "INVALID and SPECIAL characters" },
  ]

  repoTestData.forEach(
    ({ qase_id, repoName, test_explanation }) => {

      if ((qase_id === 105 || qase_id === 61)) {
        qase(qase_id,
          it(`Fleet-${qase_id}: Test GitRepo NAME with "${test_explanation}" displays warning MESSAGE and does NOT get deployed`, () => {
```
It also groups first test kind into NAME type

Reply:
Thanks for trying that out, it was new to me. I will accommodate it in my PR.